### ISSUE TITLE
[Component] [checkbox] Checkbox state is not updated when checked property changes

### ISSUE THREAD

**Issue by p1xell0rd** [element-plus/element-plus#14779]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.45`
- Element Plus Version: `2.4.2`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0 `
- Build Tool: `Vite`

### Related Component

- `el-checkbox`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdj5cbiAgICA8ZWwtY2hlY2tib3ggOmNoZWNrZWQ9XCJjaGVja2VkXCIgbGFiZWw9XCJPcHRpb24gMVwiIHNpemU9XCJsYXJnZVwiIC8+XG4gICAge3sgY2hlY2tlZCB9fVxuICA8L2Rpdj5cblxuICA8YnV0dG9uIEBjbGljaz1cImNoZWNrZWQgPSAhY2hlY2tlZFwiPnRvZ2dsZTwvYnV0dG9uPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgY2hlY2tlZCA9IHJlZihmYWxzZSlcbjwvc2NyaXB0PlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

Toggle the value of the bound ref with the button

### What is Expected?

The checkbox should be checked

### What is actually happening?

The checkbox is not checked

### Additional comments

_(empty)_

**Comment by betavs** [comment]
I believe this is normal, as the 'check' attribute only has an effect during the initialization of the component. If you need to modify the selection state, you should use the 'model-value' or 'v-model' attribute.

**Comment by CherishTheYouth** [comment]
Yeah, Maybe the documentation shoulsd indicates that checked props only effects when component initial.
